nexus/analyzer/src/lib.rs

```diff
@@ -903,5 +915,39 @@ fn parse_db_options(db_type: DbType, with_options: &[SqlOption]) -> anyhow::Resu
                 })
             }
         }
+        DbType::Mysql => Config::MysqlConfig(pt::peerdb_peers::MySqlConfig {
+            host: opts.get("host").context("no host specified")?.to_string(),
+            port: opts
+                .get("port")
+                .context("no port specified")?
+                .parse::<u32>()
+                .context("unable to parse port as valid int")?,
+            user: opts
+                .get("user")
+                .context("no username specified")?
+                .to_string(),
+            password: opts
+                .get("password")
+                .cloned()
+                .unwrap_or_default()
+                .to_string(),
+            database: opts
+                .get("database")
+                .cloned()
+                .unwrap_or_default()
+                .to_string(),
+            setup: opts
+                .get("setup")
+                .map(|s| s.split(';').map(String::from).collect::<Vec<_>>())
+                .unwrap_or_default(),
+            compression: opts
+                .get("compression")
+                .and_then(|s| s.parse::<i8>().ok())
```

Comment:
u8? also apply any restrictions if possible

Reply:
probably u32 to match mysql_async